brave/brave-core
# [ads] Do not fetch geo and catalog on iOS when Brave Rewards/News disabled

Resolves https://github.com/brave/brave-browser/issues/39621



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

* Fresh install
* Restart the Browser to make griffin seed applied
* Make sure that Brave Ads started and successfully initialized
* Make sure that there is NO network request to https://geo.ads.bravesoftware.com/v1/getstate endpoint
* Make sure that there is NO network request to https://static.ads.bravesoftware.com/v9/catalog endpoint
* Enable Brave News
* Make sure that there is network request to https://geo.ads.bravesoftware.com/v1/getstate endpoint
* Make sure that there is network request to https://static.ads.bravesoftware.com/v9/catalog endpoint
